core/types/calls.cc

```diff
@@ -46,8 +46,9 @@ DispatchResult OrType::dispatchCall(const GlobalState &gs, const DispatchArgs &a
     categoryCounterInc("dispatch_call", "ortype");
     auto leftRet = left.dispatchCall(gs, args.withSelfRef(left));
     auto rightRet = right.dispatchCall(gs, args.withSelfRef(right));
-    return DispatchResult::merge(Types::any(gs, leftRet.returnType, rightRet.returnType),
-                                 DispatchResult::Combinator::OR, std::move(leftRet), std::move(rightRet));
+    auto resultType = Types::any(gs, leftRet.returnType, rightRet.returnType);
+    return DispatchResult::merge(std::move(resultType), DispatchResult::Combinator::OR, std::move(leftRet),
+                                 std::move(rightRet));
```

Comment:
I assume this was done for clarity and for symmetry with the `AndType` case?  +1 to pulling out some temporary variables to make the code clearer.

Reply:
It was done to avoid potentially relying on a moved value, but also the clarity is good :)